zenoh/src/api/handlers/fifo.rs

```diff
@@ -38,7 +38,11 @@ impl<T: Send + 'static> IntoHandler<'static, T> for FifoChannel {
     type Handler = flume::Receiver<T>;
 
     fn into_handler(self) -> (Callback<'static, T>, Self::Handler) {
-        flume::bounded(self.capacity).into_handler()
+        if self.capacity == 0 {
```

Comment:
I don't like implicit behaviour with magic numbers that change the type of the internal channel.
Users can already pass an unbound channel with 
```rust
session
    .declare_subscriber("myke")
    .with(flume::unbounded())
    .await
    .unwrap();
```

For the above reason I don't see the need of such change.